### ISSUE TITLE
How to call function in repository from another repository

### ISSUE THREAD

**Issue by Firman95** [freevital/laravel-repository#2]
```
class ARepository extends BaseRepository {

	public function model()
	{
		return "App\Models\A";
	}

	public function createNewA(){
		DB::beginTransaction();
		try{
	            $this->model->create([
	            'column_a' => 'test'
	            ]);
		    DB::commit();
		 }catch(Exception $e){
		      DB::rollback();
		    }
		 }
  }
```

How to call function createNewA() in another repo? I tried this but it did not work

```
class BRepository extends BaseRepository {

  protected $aRepo;

  public function model()
  {
      return "App\Models\B";
  }

  public function createNewB(){
    $this->aRepo = new ARepository();
    DB::beginTransaction();
    try{
      $this->aRepo->createNewA();
      $this->model->create([
        'column_b' => 'test'
      ]);
      DB::commit();
    }catch(Exception $e){
      DB::rollback();
    }
  }
}
```

This is the controller

```
class HomeController extends Controller
{
    protected $repo;
    public function __construct(BRepository $repo)
    {
         $this->repo = $repo;
         $this->repo->createNewB();
    }
}
```

**Comment by Firman95** [comment]
Solved. This is how i declare repository in repository `$this->aRepo = \App::make(ARepository::class);`

```
class BRepository extends BaseRepository {

  protected $aRepo;

  public function model()
  {
      return "App\Models\B";
  }

  public function createNewB(){
    $this->aRepo = \App::make(ARepository::class);
    DB::beginTransaction();
    try{
      $this->aRepo->createNewA();
      $this->model->create([
        'column_b' => 'test'
      ]);
      DB::commit();
    }catch(Exception $e){
      DB::rollback();
    }
  }
}
```

**Comment by freevital** [comment]
You may inject another repository in construct method:
```php
class BRepository extends BaseRepository
{
    /**
     * @var ARepository
     */
    private $aRepo;

    /**
     * @param Application $app
     * @param ARepository $aRepo
     */
    public function __construct(Application $app, ARepository $aRepo)
    {
        parent::__construct($app);

        $this->aRepo = $aRepo;
    }
}
```

**Comment by Firman95** [comment]
@freevital 
If I inject BRepo in ARepo, how to inject ARepo in BRepo?

This is my BRepo
```
class BRepository extends BaseRepository
{
    /**
     * @var ARepository
     */
    private $aRepo;

    /**
     * @param Application $app
     * @param ARepository $aRepo
     */
    public function __construct(Application $app, ARepository $aRepo)
    {
        parent::__construct($app);

        $this->aRepo = $aRepo;
    }
}
```
This is my ARepo
```
class ARepository extends BaseRepository
{
    /**
     * @var BRepository
     */
    private $bRepo;

    /**
     * @param Application $app
     * @param BRepository $bRepo
     */
    public function __construct(Application $app, BRepository $bRepo)
    {
        parent::__construct($app);
        $this->bRepo = $bRepo;
    }
}
```
I get Http Error 500, "domain.com is currently unable to handle this request", for all routing that use ARepository or BRepository class. I can inject ARepository with all other repositories with no problem except for BRepository class. Thanks in advance.

**Comment by freevital** [comment]
#3 Duplicate

**Comment by freevital** [comment]
#3

**Comment by freevital** [comment]
closes #3
